hastory-data-gen/src/Data/Hastory/Gen.hs

```diff
@@ -29,7 +30,9 @@ instance GenValid UserForm where
   shrinkValid = shrinkValidStructurallyWithoutExtraFiltering
 
 instance GenValid Username where
-  genValid = mkUsername <$> (genValid `suchThat` notNull)
+  genValid =
+    mkUsername <$>
+    (genValid `suchThat` \rawUsername -> notNull rawUsername && T.all isAlphaNum rawUsername)
```

Comment:
This works, but it's quite slow.
You'll want to make a generator benchmark before you improve it though.
See genvalidity-criterion.

Reply:
How do you feel about an issue on this - #31 
? I'd love to tackle it (since I have never done benchmarking on Haskell and think I can learn a lot), but you know me - always trying to keep focus. 

If you feel otherwise, let me know and I'll delete the issue and address here.